homeassistant/components/homekit_controller/air_quality.py

```diff
@@ -0,0 +1,110 @@
+"""Support for HomeKit Controller air quality sensors."""
+from homekit.model.characteristics import CharacteristicsTypes
+
+from homeassistant.components.air_quality import AirQualityEntity
+
+from . import KNOWN_DEVICES, HomeKitEntity
+
+AIR_QUALITY_TEXT = {
+    0: "unknown",
+    1: "excellent",
+    2: "good",
+    3: "fair",
+    4: "inferior",
+    5: "poor",
+}
+
+
+class HomeAirQualitySensor(HomeKitEntity, AirQualityEntity):
+    """Representation of a HomeKit Controller Air Quality sensor."""
+
+    def get_characteristic_types(self):
+        """Define the homekit characteristics the entity cares about."""
+        return [
+            CharacteristicsTypes.AIR_QUALITY,
+            CharacteristicsTypes.DENSITY_PM25,
+            CharacteristicsTypes.DENSITY_PM10,
+            CharacteristicsTypes.DENSITY_OZONE,
+            CharacteristicsTypes.DENSITY_NO2,
+            CharacteristicsTypes.DENSITY_SO2,
+            CharacteristicsTypes.DENSITY_VOC,
+        ]
+
+    @property
+    def particulate_matter_2_5(self):
+        """
+        Return the particulate matter 2.5 level.
+
+        This is required on the HA side but optional on the HomeKit side, so we
+        always return a value.
+        """
+        return self.get_hk_char_value(CharacteristicsTypes.DENSITY_PM25) or 0
```

Comment:
We could return `None` to indicate unknown value. Isn't that better than 0?

Reply:
If i return `None` then the platform will omit the value entirely, but it is a required attribute.

Reply:
It's required to implement this property. It's not required to return not `None`.

The state will still read this property and then report unknown state. Yes, this item will not be included in state attributes, but that's according to our standard. We don't include unknown value items in state attributes.

Reply:
Ok, thank you for the clarification! This is now done too.